rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/internal/ExtraActionPickerDialogFragment.java

```diff
@@ -0,0 +1,62 @@
+package chat.rocket.android.widget.internal;
+
+import android.app.Dialog;
+import android.support.design.widget.BottomSheetDialogFragment;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.List;
+import chat.rocket.android.widget.R;
+import chat.rocket.android.widget.layouthelper.MessageExtraActionListAdapter;
+import chat.rocket.android.widget.message.MessageExtraActionItemPresenter;
+
+public class ExtraActionPickerDialogFragment extends BottomSheetDialogFragment {
+
+  private List<MessageExtraActionItemPresenter> actionItems;
+  private Callback callback;
+
+  public static ExtraActionPickerDialogFragment create(
+      List<MessageExtraActionItemPresenter> actionItems, Callback callback) {
+    ExtraActionPickerDialogFragment fragment = new ExtraActionPickerDialogFragment();
+    fragment.setActionItems(actionItems);
+    fragment.setCallback(callback);
+
+    return fragment;
+  }
+
+  public void setActionItems(List<MessageExtraActionItemPresenter> actionItems) {
+    this.actionItems = actionItems;
+  }
+
+  public void setCallback(Callback callback) {
+    this.callback = callback;
```

Comment:
Does this work when Fragment is re-instantiated by Android framework...?

I'm afraid that I don't know the detail, but it seems that Google doesn't recommend creating setter methods in Fragment for setting custom callbacks.
https://developer.android.com/training/basics/fragments/communicating.html